### PR TITLE
#1082 Documents tab

### LINKED ISSUE #1082: New Review home page: Documents tab
Implementation details (from https://docs.google.com/document/d/1LWCvv42dHwcv4yCWKG26MJI8yZ6QGHAnSwpY7-rWblQ/)

### Back-end:

(Done in PR: https://github.com/openmsupply/application-manager-server/pull/646)

- Add field “is_output_doc” to “files” table (boolean)
- Add field “description” to “files” table (string)
- Update “Generate Document” Action to (optionally) add these fields when creating docs
- Update existing templates to use where appropriate

### Front-end:

- ~~Update fileUpload plugin to have optional "description" parameter and display it in both Application and Summary view~~
- Query all files for current application where “is_output_doc = TRUE”. 
- Display as thumbnails and filename labels, with each linking to document (opens in new tab) (using fileUpload component Summary View – need to add “description” field to it)

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fix #1082

Front-end elements for displaying application documents. I didn't end up using the SummaryView of the fileUpload plugin, as wrangling the data to suit the input requirements would be more complicated than just displaying the docs in a simple list.

You won't see anything unless the "is_output_doc" field on files is set to `true`, so you can use the snapshot `application-manager-templates/dev/snapshots/documents_tab_2022-02-03` with a couple already set. The page to view is: http://localhost:3000/application/S-JPH-0100/review?tab=documents

Screen Shot 2022-02-03 at 12 15 33 PM

One visual issue is that the "generic" icons (PDF, etc.) look fine as is, but the ones with auto-generated thumbnails could do with a light border to set them apart from the background. However, putting a border on it messes up the look of the generic icons. I figured that most (all?) of the generated docs are going to be PDFs, so I've left the border off for now.

### CODE REVIEW

**Comment by CarlosNZ** [comment]
Things to add to this PR from today's meeting:
- [ ] Show Date for each document
- [ ] (Try) grey background block to make document thumbnails look less crappy

**Comment by CarlosNZ** [comment]
> * [x]  Show Date for each document
> 
> * [x] (Try) grey background block to make document thumbnails look less crappy

Looking better now, I think:
Screen Shot 2022-02-05 at 10 28 19 PM

I don't love the grey background, but can't think of anything better for now that's consistent with the overall UI look.

**Review by nmadruga — COMMENTED** [review]
I'm trying to test this while doing changes to PR https://github.com/openmsupply/application-manager-server/pull/651 I finally got to a point where I can run the actioins but it's failing in `modify record`. I'll commit my changes to the snapshot to have the 2 updates templates (User registration and Grant user permission) to be using null when no orgId should be linked and change the permission to review the "Sample template" from reviewGeneral - which isn't associated to Admin user to use the reviewScrening permission so the Admin can review that one.

File: src/containers/Review/documents/DocumentsTab.tsx [hunk 1]
```diff
@@ -1,25 +1,65 @@
...
+            <div className="flex-row-start flex-extra">
+              <div className="icon-container">
+                <Image src={`${downloadUrl}/file?uid=${doc?.uniqueId}&thumbnail=true`} />
+              </div>
+              <div>
+                <Header as="h4" content={doc?.description} />
+                <p className="slightly-smaller-text">
+                  {DateTime.fromISO(doc?.timestamp).toLocaleString()}
+                </p>
+                {/* <p>{doc?.description}</p> */}
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply src/containers/Review/documents/DocumentsTab.tsx]: Oops, leftover from from my old version. Removed now 16eb0152

File: src/containers/Review/documents/DocumentsTab.tsx [hunk 2]
```diff
@@ -1,25 +1,65 @@
...
+                <p className="smaller-text">
+                  <a href={`${downloadUrl}/file?uid=${doc?.uniqueId}`} target="_blank">
+                    {doc?.originalFilename}
+                  </a>
+                </p>
+                {/* <p>{DateTime.fromISO(doc?.timestamp).toLocaleString()}</p> */}
```
[the remark this answers is not in the document]
Comment (CarlosNZ) [reply src/containers/Review/documents/DocumentsTab.tsx]: Same 16eb0152

**Review by nmadruga — APPROVED** [review]
LGTM

**Pull request merged**